### ISSUE TITLE
MinimumTapTargetGuideline doesn't handle elements that are scrolled but not at the edge of the screen

### ISSUE THREAD

**Issue by jacobsimionato** [flutter/flutter#108060]
Cross posting from internal issue 238879865:

MinimumTapTargetGuideline has logic to detect and skip elements that are scrolled partially off screen. However, this logic is flawed. It only checks for widgets that are at the edge of the physical screen, which omits the common case where the Scrollable container itself doesn't touch the edge of the screen.

Consider a screen with an AppBar and a long scrollable widget beneath it. Something like:
```
Scaffold(
  appBar: AppBar(title: Text('Foo')),
  body: ListView(children: buildChildren()),
)
```

If a widget is partially visible at the top of the ListView, it will still be beneath the AppBar, not at the edge of the device screen, therefore it may fail the MinimumTapTargetGuideline erroneously.

A proposed fix in MinimumTapTargetGuideline in accessibility.dart (see internal commit 460734097):
```
Evaluation _traverse(WidgetTester tester, SemanticsNode node) {
    Evaluation result = const Evaluation.pass();
    node.visitChildren((SemanticsNode child) {
      result += _traverse(tester, child);
      return true;
    });
    if (node.isMergedIntoParent) {
      return result;
    }
    if (shouldSkipNode(node)) {
      return result;
    }
    Rect paintBounds = node.rect;
    SemanticsNode? current = node;
    // **Added logic**
    while (current != null &&
        !current.hasFlag(SemanticsFlag.hasImplicitScrolling)) {
      final Matrix4? transform = current.transform;
      if (transform != null) {
        paintBounds = MatrixUtils.transformRect(transform, paintBounds);
      }
      current = current.parent;
    }
    // skip node if it is touching the edge of the screen, since it might
    // be partially scrolled offscreen.
    const double delta = 0.001;
    // **Added logic**
    final Size physicalSize =
        current?.rect.size ?? tester.binding.window.physicalSize;
    if (paintBounds.left <= delta ||
        paintBounds.top <= delta ||
        (paintBounds.bottom - physicalSize.height).abs() <= delta ||
        (paintBounds.right - physicalSize.width).abs() <= delta) {
      return result;
    }
    // shrink by device pixel ratio.
    final Size candidateSize = paintBounds.size / tester.binding.window.devicePixelRatio;
    if (candidateSize.width < size.width - delta ||
        candidateSize.height < size.height - delta) {
      result += Evaluation.fail(
        '$node: expected tap target size of at least $size, '
        'but found $candidateSize\n'
        'See also: $link',
      );
    }
    return result;
```

**Comment by jacobsimionato** [comment]
Duplicate with https://github.com/flutter/flutter/issues/107615 - sorry!
